datasync/downloader/queue.go

```diff
@@ -325,19 +309,19 @@ func (q *queue) Schedule(headers []*types.Header, from uint64) []*types.Header {
 		// Make sure no duplicate requests are executed
 		if _, ok := q.blockTaskPool[hash]; ok {
 			logger.Trace("Header already scheduled for block fetch", "number", header.Number, "hash", hash)
-			continue
-		}
-		if _, ok := q.receiptTaskPool[hash]; ok {
-			logger.Trace("Header already scheduled for receipt fetch", "number", header.Number, "hash", hash)
-			continue
```

Comment:
The `continue` is removed, but the new fetch won't happen since `else` case is added. This change updates `from ` and `headerHead` variable in order to calculate them correctly.

Reply:
Because of the `continue`, header was not appended to `inserts` every time, however, now it is always appended to `inserts`. Is it okay?

Reply:
@ehnuje Right. Previously, it is considered as a bad peer if a body request exists. I think ignoring duplicated body requests is fine.